pkgs/tools/compression/xz/xz-after-bootstrap.nix

```diff
@@ -0,0 +1,85 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  automake,
+  autoconf,
+  perl,
+  libtool,
+  gettext,
+  perl538Packages,
+  xz,
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "xz";
+  version = xz.version;
+
+  src = fetchFromGitHub {
+    owner = "tukaani-project";
+    repo = "xz";
+    rev = "v${finalAttrs.version}";
+    hash = "sha256-Xp1uLtQIoOG/qVLpq5D/KFmTOJ0+mNkNclyuJsvlUbE=";
+  };
+
+  strictDeps = true;
+
+  outputs = [
+    "bin"
+    "dev"
+    "out"
+    "man"
+    "doc"
+  ];
+
+  enableParallelBuilding = true;
+
+  doCheck = true;
+
+  nativeBuildInputs = [
+    gettext
+    libtool
+    automake
+    autoconf
+    perl
+    perl538Packages.Po4a
+  ];
+
+  preConfigure = ''
+    ./autogen.sh
+  '';
+
+  postPhases = [ "compareArtifacts" ];
+
+  compareArtifacts = ''
+    diff $out/lib/liblzma.so ${xz.out}/lib/liblzma.so
+  '';
```

Comment:
Maybe we can write a Nix function that gets a derivation as a single input, and iterates all `.so` files in it and compares them to the `.so` files in `$out`.. That should be a fairly good starting point in generalizing this.

Reply:
There is also something that I experienced in my testing (again to keep things simple for this discussion I did not include it), is that sometimes packages have self references and we have to get rid of them before comparing the two versions.